source/client.c

```diff
@@ -430,38 +469,49 @@ struct aws_mqtt_client_connection *aws_mqtt_client_connection_new(struct aws_mqt
     /* Initialize the client */
     connection->allocator = client->allocator;
     connection->client = client;
-    connection->state = AWS_MQTT_CLIENT_STATE_DISCONNECTED;
+    connection->synced_data.state = AWS_MQTT_CLIENT_STATE_DISCONNECTED;
```

Comment:
Should probably zero out the entire synced_data structure before we start filling in values